### PR TITLE
Docs: dogfood Gestalt ESLint rules on our docs

### PR DESCRIPTION
**Pull request by christianvuerings**
Let's [dogfood](https://en.wikipedia.org/wiki/Eating_your_own_dog_food) our ESLint rules on our own docs:

* Helps with spotting issues with our rules
* Ensures our docs are up to the highest coding standards (perhaps this is an exaggeration) 

## Test Plan

* ESLint passes in CI
* Modified docs pages still look good

### CODE REVIEW

File: docs/src/components/FeedbackCallout.js [hunk 1]
```diff
@@ -14,26 +14,24 @@ export default function FeedbackCallout({ componentName }: Props): Node {
   const link = `${BASE_LINK}${componentName}`;
 
   return (
-    <Box marginLeft={-1} marginRight={-1}>
```
Comment (christianvuerings) [inline docs/src/components/FeedbackCallout.js:17]: `marginLeft` and `marginRight` are no longer supported on `Box` so this didn't do anything

File: packages/eslint-plugin-gestalt/src/button-icon-restrictions.js [hunk 2]
```diff
@@ -1,6 +1,5 @@
 /**
  * @fileoverview Require specific props when using an icon with Button
- * @author Ryan James <rjames@pinterest.com>
```
Comment (christianvuerings) [inline packages/eslint-plugin-gestalt/src/button-icon-restrictions.js:3]: Removed all authors, no reason to make these public

File: packages/eslint-plugin-gestalt/src/button-icon-restrictions.js [hunk 3]
```diff
@@ -50,7 +49,7 @@ const rule = {
           return;
         }
         importedComponent = decl.specifiers.some((node) => {
-          const isValidComponent = node.imported.name === componentName;
+          const isValidComponent = node.imported?.name === componentName;
```
Comment (christianvuerings) [inline packages/eslint-plugin-gestalt/src/button-icon-restrictions.js:52]: Discovered an issue with a prop called `name` on the `Example.js` page:

https://github.com/pinterest/gestalt/blob/fe1aa734f729e9ce5b249b0b820c7d466a180e18/docs/src/components/Example.js#L40

File: docs/src/components/FeedbackCallout.js (hunk 1 above)
Comment (dangerismycat) [reply docs/src/components/FeedbackCallout.js:17]: weird that Flow wasn't complaining about this given that we removed those props a while ago…

**Review by dangerismycat — APPROVED** [review]
Awesome, thanks Christian!

**Comment by dangerismycat** [comment]
Closed without merging?

**Pull request closed without merging**